frontend/src/views/Storage/Storage.tsx

```diff
@@ -10,66 +10,86 @@ import {
   ButtonGroup,
 } from '@bcgov/design-system-react-components';
 import { DataGrid, GridColDef, GridRenderCellParams } from '@mui/x-data-grid';
-import { NMPFileFarmManureData } from '../../types';
-import { DEFAULT_NMPFILE_YEAR, DefaultManureFormData } from '../../constants';
 import { StyledContent } from './storage.styles';
-import useAppState from '../../hooks/useAppState';
 import { NUTRIENT_ANALYSIS, MANURE_IMPORTS } from '../../constants/routes';
 
 import { AppTitle, PageTitle, ProgressStepper, TabsMaterial } from '../../components/common';
 import { addRecordGroupStyle, customTableStyle, tableActionButtonCss } from '../../common.styles';
 import StorageModal from './StorageModal';
+import useAppState from '@/hooks/useAppState';
+import { NMPFileManureStorageSystemsData } from '@/types';
+import DefaultNMPFileManureStorageSystemsData from '@/constants/DefaultNMPFileManureStorage';
 
 export default function Storage() {
-  const { state } = useAppState();
+  const { state, dispatch } = useAppState();
   const navigate = useNavigate();
-
-  // TODO: make correct file type
-  const [storageList, setStorageList] = useState<NMPFileFarmManureData[]>(
-    state.nmpFile.years[0]?.FarmManures || [],
+  const [rowEditIndex, setRowEditIndex] = useState<number | undefined>(undefined);
+  const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
+  const [storageList, setStorageList] = useState<NMPFileManureStorageSystemsData[]>(
+    state.nmpFile.years[0]?.ManureStorageSystems || [],
+  );
+  const [storageFormData, setStorageFormData] = useState<NMPFileManureStorageSystemsData>(
+    DefaultNMPFileManureStorageSystemsData,
```

Comment:
`storageFormData` should be a stateful variable in the StorageModal component, not this component. You can see how we handle CropsModal or AddAnimalsModal for examples.